src/luaclang.c

```diff
@@ -1,14 +1,367 @@
+{
+        CXCursor *c = (CXCursor*) luaL_checkudata(L, n, CURSOR_METATABLE);
+        return *c;
+}
+
+/* Create CXType as userdata */
+
+static CXType * new_CXType(lua_State *L)
+{
+        CXType *t = (CXType*) lua_newuserdata(L, sizeof(CXType));
+        luaL_getmetatable(L, TYPE_METATABLE);
+        lua_setmetatable(L, -2);
+        return t;
+}
+
+
+/* --Clang function-- */
+
+/*      
+        Format - createIndex(exclude_pch, diagnostics)
+        Parameters - 1. exclude_pch - When non-zero, allows enumeration of "local" declarations
+                     2. diagnostics - Display diagnostics
+        More info - https://clang.llvm.org/doxygen/group__CINDEX.html#ga51eb9b38c18743bf2d824c6230e61f93
+*/
+static int create_CXIndex(lua_State *L) 
+{
+        int exclude_pch = lua_toboolean(L, 1);
+        int diagnostics = lua_toboolean(L, 2);
+        CXIndex *idx = new_CXIndex(L);
+        *idx = clang_createIndex(exclude_pch, diagnostics);
+        return 1;
+}
+
+static luaL_Reg clang_function[] = {
+        {"createIndex", create_CXIndex},
+        {NULL, NULL}
+};
+
+
+/* --Index functions-- */
+
+/*      
+        Format - idx:disposeIndex()
+        Parameter -  idx - Index to be destroyed       
+        More info - https://clang.llvm.org/doxygen/group__CINDEX.html#ga166ab73b14be73cbdcae14d62dbab22a
+*/
+static int dispose_CXIndex(lua_State *L) 
+{
+        CXIndex idx = to_CXIndex(L, 1);
+        luaL_argcheck(L, idx != NULL, 1, "arg disposed");
+        clang_disposeIndex(idx);
+        idx = NULL;
+        return 0;
+}
+
+/*      
+        Format - idx:parseTU(file_name)
+        Parameters - 1. idx - The index object with which the translation unit will be associated  
+                     2. file_name - The name of the source file to load    
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#ga2baf83f8c3299788234c8bce55e4472e
+*/
+static int parse_TU(lua_State *L)
+{
+        CXIndex idx = to_CXIndex(L, 1);
+        const char *file_name = lua_tostring(L, 2);
+        const char *args[] = {file_name};
+        CXTranslationUnit *tu = new_CXTU(L);
+        *tu = clang_parseTranslationUnit(idx, 0, args, 1, 0, 0, CXTranslationUnit_None);
+        return 1;
+}
+
+
+static luaL_Reg index_functions[] = {
+        {"disposeIndex", dispose_CXIndex},
+        {"parseTU", parse_TU},
+        {"__gc", dispose_CXIndex},
+        {NULL, NULL}
+};
+
+
+/* --Translation unit functions-- */
+
+/*      
+        Format - tu:disposeTU()
+        Parameter - tu - Translation unit to be destroyed       
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#gaee753cb0036ca4ab59e48e3dff5f530a
+*/
+static int dispose_CXTU(lua_State *L) 
+{
+        CXTranslationUnit tu = to_CXTU(L, 1);
+        luaL_argcheck(L, tu != NULL, 1, "arg disposed");
+        clang_disposeTranslationUnit(tu);
+        tu = NULL;
+        return 0;
+}
+
+/*      
+        Format - tu:getTUCursor()
```

Comment:
You can drop the class name (`TU`) from the function's name since it is a method now: `getCursor` is enough.
Please do so to all methods: e.g. `getCursorSpelling` -> `getSpelling`.